cmd/operator-sdk/bundle/validate.go

```diff
@@ -96,22 +105,24 @@ To build and validate an image:
 				}
 				defer func() {
 					if err = os.RemoveAll(c.directory); err != nil {
-						log.Errorf("Error removing temp bundle dir: %v", err)
+						logger.Errorf("Error removing temp bundle dir: %v", err)
 					}
 				}()
 
-				log.Info("Unpacking image layers")
-
+				logger.Info("Unpacking image layers")
 				if err := c.unpackImageIntoDir(args[0], c.directory); err != nil {
-					log.Fatalf("Error unpacking image %s: %v", args[0], err)
+					logger.Fatalf("Error unpacking image %s: %v", args[0], err)
 				}
 			}
 
-			if err = c.run(); err != nil {
-				log.Fatal(err)
+			result := c.run()
+			if result.Passed { // if no errors were added
```

Comment:
Since you're already in a passing state, no need to also log a message. Also saying "all validation tests have completed successfully" isn't true if there are warnings.

Reply:
But will not true and very confuse mainly if has errors. Do not you think?

Reply:
If it has errors or warnings, there should not be a message saying "all validation passed" since it technically didn't pass. The consumer should see
```
{"passed": true, "output": []}
```
or
```
{"passed": true, "output": [{"type": "warning", msg: "..."}]}
```
and know the validation has passed.

Reply:
Make sense 👍